addons/survey/views/survey_templates_print.xml

```diff
@@ -34,12 +34,64 @@
                                             <span t-if="question.constr_mandatory" class="text-danger">*</span>
                                             <span t-if="scoring_display_correction" class="badge badge-pill" t-att-data-score-question="question.id"></span>
                                         </h2>
-                                        <t t-if="question.description"><div class="text-muted oe_no_empty" t-field='question.description'/></t>
-                                        <t t-if="question.question_type == 'text_box'"><t t-call="survey.question_text_box"/></t>
-                                        <t t-if="question.question_type == 'char_box'"><t t-call="survey.question_char_box"/></t>
-                                        <t t-if="question.question_type == 'numerical_box'"><t t-call="survey.question_numerical_box"/></t>
-                                        <t t-if="question.question_type == 'date'"><t t-call="survey.question_date"/></t>
-                                        <t t-if="question.question_type == 'datetime'"><t t-call="survey.question_datetime"/></t>
+                                        <t t-if="not is_html_empty(question.description)"><div class="text-muted oe_no_empty mt-1" t-field='question.description'/></t>
+                                        <t t-if="question.question_type in ['numerical_box', 'date', 'datetime', 'text_box', 'char_box']">
```

Comment:
This whole part is honestly a bit hard to read.
I suggest we rework it a bit:

```xml
<t t-if="question.question_type in ['numerical_box', 'date', 'datetime', 'text_box', 'char_box']">
    <t t-set="answer_line" t-value="answer_lines[0]"/>
    <t t-if="answer_line.skipped">
        <!-- question was skipped, display an orange block with the text "Skipped" in place of the answer -->
        <!-- rest of the code to display the skipped block here, should be identical for all question types -->
    </t>
    <t t-else="">
        <!-- rest of the code to compute the bg color of the answer block and then call the appropriate template -->
        <!-- (would be nice to remove the "survey_text_fields" and "survey_scored_fields" template if only called once)-->
    </t>
    <t t-if="question.question_type in ['numerical_box', 'date', 'datetime'] and question['answer_%s' % question.question_type] and not answer_line.answer_is_correct">
        <!-- User did not pick the correct answer, let's show it to him -->
    </t>
</t>
```

Reply:
Hello @awa-odoo ,
Done!
I have done some other changes too, which will cover the use-case
1) when the survey is with a no-scoring type.
2) in a live survey if a user was not able to submit any answer in any question at that time we don't get answer_lines.